engine/src/io/data_parser/sql/PostgreSQLParser.cpp

```diff
@@ -0,0 +1,549 @@
+        const std::int64_t hostOrderedValue = ntohl(castedValue);
+        std::vector<std::uint64_t> &vector =
+            *reinterpret_cast<std::vector<std::uint64_t> *>(
+                host_cols[projection_index]);
+        vector.push_back(hostOrderedValue);
+        break;
+      }
+      case cudf::type_id::FLOAT32:
+      case cudf::type_id::DECIMAL32: {
+        const std::int32_t castedValue =
+            *reinterpret_cast<const std::int32_t *>(resultValue);
+        const std::int32_t hostOrderedValue = ntohl(castedValue);
+        const float floatCastedValue =
+            *reinterpret_cast<const float *>(&hostOrderedValue);
+        std::vector<float> &vector = *reinterpret_cast<std::vector<float> *>(
+            host_cols[projection_index]);
+        vector.push_back(floatCastedValue);
+        break;
+      }
+      case cudf::type_id::FLOAT64:
+      case cudf::type_id::DECIMAL64: {
+        const std::int64_t castedValue =
+            *reinterpret_cast<const std::int64_t *>(resultValue);
+        const std::int64_t hostOrderedValue = ntohl(castedValue);
+        const double doubleCastedValue =
+            *reinterpret_cast<const double *>(&hostOrderedValue);
+        std::vector<double> &vector = *reinterpret_cast<std::vector<double> *>(
+            host_cols[projection_index]);
+        vector.push_back(doubleCastedValue);
+        break;
+      }
+      case cudf::type_id::BOOL8: {
+        const std::uint8_t castedValue =
+            *reinterpret_cast<const std::uint8_t *>(resultValue);
+        std::vector<std::uint8_t> &vector =
+            *reinterpret_cast<std::vector<std::uint8_t> *>(
+                host_cols[projection_index]);
+        vector.push_back(castedValue);
+        break;
+      }
+      case cudf::type_id::STRING: {
+        cudf_string_col *string_col =
+            reinterpret_cast<cudf_string_col *>(host_cols[projection_index]);
+        if (isNull) {
+          string_col->offsets.push_back(string_col->offsets.back());
+        } else {
+          std::string data(resultValue);
+          string_col->chars.insert(
+              string_col->chars.end(), data.cbegin(), data.cend());
+          string_col->offsets.push_back(string_col->offsets.back() +
+                                        data.length());
+        }
+        break;
+      }
+      default: throw std::runtime_error("Invalid cudf type id");
+      }
+      if (isNull) {
+        cudf::set_bit_unsafe(null_masks[projection_index].data(), i);
+      }
+    }
+  }
+
+  cudf::io::table_with_metadata tableWithMetadata;
+  std::vector<std::unique_ptr<cudf::column>> cudf_columns;
+  cudf_columns.resize(static_cast<std::size_t>(resultNfields));
+  for (const std::size_t projection_index : column_indices) {
+    cudf::type_id cudf_type_id = cudf_types[projection_index];
+    switch (cudf_type_id) {
+    case cudf::type_id::INT8: {
+      std::vector<std::int8_t> *vector =
```

Comment:
this whole pattern with the void * feels fragile i wonder if just managing the memory and types is better

Reply:
totally agree, we already did that, our first approach was using typed vectors, but the casting was an issue, also the handling for strings is different.
Let me try again and see if we can get ride of void*

Reply:
@felipeblazing @williamBlazing If we get ride of bool we would need to iterate all the rows later to have the references for cudf, I think `void*` is not ideal but is what it offers the best performance for now (i.e. it minimizes a lot the travels over all the rows, at least 2 times: one for getting the refs and the other for adapting the refs for the cudf column ctors) cc @gcca